src/puppetlabs/trapperkeeper/authorization/acl.clj

```diff
@@ -44,57 +46,58 @@
 
 ;; ACE creation
 
-(defn munge-name
-  [pattern]
-  (-> pattern (str/lower-case) (str/split #"\.") reverse vec))
+(schema/defn ^:private split-domain :- [String]
+  "Given a domain string, split it on '.' and reverse it. For examples,
+  'sylvia.plath.net' becomes ['net' 'plath' 'sylvia']. This is used for domain
+  matching."
+  [pattern :- String]
+  (-> pattern
+      (clojure.string/lower-case)
+      (clojure.string/split #"\.")
+      reverse
+      vec))
```

Comment:
This is old code, I just left it in. Perhaps the original author thought reverse was lazy, but it's not; I'll take it out.